pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java

```diff
@@ -2764,51 +2764,62 @@ private PositionImpl calculatePositionAckSet(boolean isExcluded, int batchSize,
     protected void internalGetMessageById(AsyncResponse asyncResponse, long ledgerId, long entryId,
                                               boolean authoritative) {
         // will redirect if the topic not owned by current broker
-        validateTopicOwnershipAsync(topicName, authoritative)
-                .thenCompose(__ -> validateTopicOperationAsync(topicName, TopicOperation.PEEK_MESSAGES))
-                .thenCompose(__ -> {
-                    CompletableFuture<Void> ret;
-                    if (topicName.isGlobal()) {
-                        ret = validateGlobalNamespaceOwnershipAsync(namespaceName);
+        getPartitionedTopicMetadataAsync(topicName, authoritative, false)
+                .thenAccept(partitionMetadata -> {
+                    if (!topicName.isPartitioned() && partitionMetadata.partitions > 0) {
```

Comment:
Need to remove '!topicName.isPartitioned()'

Reply:
done. @Technoboy-

Reply:
This same condition still exists in this class. I will fix it in another pr.